games/guess-it/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+            auth.drop();
+            // Secure the funds
+            self.bank.put(payment.take(Decimal::from(self.game.bet.to_string())));
+            // Add the player's details
+            self.game.players.insert(nft_id.clone(), player.clone());
+            self.game.state = if self.game.players.len() == 2 { State::MakeGuess } else { State::AcceptingChallenger };
+
+            return (badge, payment, nft_id.to_string());
+        }
+
+        pub fn state(&mut self) -> String {
+            Game::serialize(&self.game)
+        }
+
+        #[auth(badge_ref)]
+        pub fn make_guess(&mut self, guess: String) -> String {
+            assert_eq!(self.game.state, State::MakeGuess, "Not ready to make a guess!");
+
+            let key = auth.get_non_fungible_key();
+            let player = self.game.players.get_mut(&key).unwrap();
+            player.make_guess(guess.clone());
+
+            let guesses: Vec<String> = self.game.players.clone().into_iter()
+                .map(|(_key, player)| player.get_guess())
+                .filter(|guess| !guess.is_empty())
+                .collect();
+            self.game.state = if guesses.len() == 2 { State::WinnerSelection } else { State::MakeGuess };
+            if self.game.state == State::WinnerSelection {
+                let response = self.check_winner();
+                if self.game.winner != NonFungibleKey::from(0).to_string() {
+                    self.game.state = State::Payout;
+                }
+                format!("Your guess '{}' was accepted\nResponse: {}", guess, response)
+            } else {
+                format!("Your guess '{}' was accepted", guess)
+            }
+
+        }
+
+        pub fn check_winner(&mut self) -> String {
+            // Pick random number and store it
+            let random_number = (get_random(&self.game) % 6) + 1;
+            // determine who was closest without going over
+            self.game.winner = get_winner(&self.game, random_number);
+            self.game.last_roll = random_number.to_string();
+            // update player's share of pot
+            format!("Dice roll was: '{}' .... winner is: {}",random_number, self.game.winner)
+        }
+
+        #[auth(badge_ref)]
+        pub fn withdraw_funds(&mut self) -> (Bucket, String) {
+            assert_eq!(self.game.state, State::Payout, "It's not time to get paid!");
+            let nft_id: NonFungibleKey = auth.get_non_fungible_key();
+
+            if nft_id.to_string() == self.game.winner {
+                let amt = self.bank.amount();
+                let payout = self.bank.take(amt);
+
+                (payout, format!("You withdrew {:?} XRD!", amt))
+            }
+            else { panic!("You cannot withdraw funds!") }
+        }
+
+        pub fn get_context(&self) -> (Actor, Address, H256, u64, u128) {
+            context::ContextTest::query()
+        }
+    }
+}
+
+fn get_random(game: &Game) -> u128 {
```

Comment:
There are a few things about this `get_random` function:
1- This is a pseudorandom function which is okay for an example but not for a full working product. It's okay to keep it for now and there is no need to change it as this is just an example.

2- Your `multiplier` is always 0 which means that the `get_random` function always returns 0. What is the purpose of the `multiplier` in this case?

Reply:
The multiplier should be just p1's guess and p2's guess added together and then multiplied by the modulus operation on the Uuid.

it was just a quick way to add some more randomness to the RNG.

Reply:
To be on the safe side of this, I'll be adding a test to make sure the multiplier isn't 0
<img width="975" alt="image" src="https://user-images.githubusercontent.com/7891291/164724621-d69033f9-1936-43d7-9767-640dc7b195ff.png">

Reply:
The major issue I didn't notice until adding tests was that the UUID could not be multiplied, as it overflows the u128 bounds. Instead of multiplying, dividing produced the desired effect of increasing the randomness of the roll.

If there's any other way to determine this without relying on oracles, I'm willing to implement it. 

<img width="919" alt="image" src="https://user-images.githubusercontent.com/7891291/164730162-51b1f956-53d8-4b4d-9552-95436b16cd47.png">